params.go

```diff
@@ -314,6 +314,12 @@ func BuildQueryString(opts interface{}) (*url.URL, error) {
 						params.Add(tags[0], strconv.FormatInt(v.Int(), 10))
 					case reflect.Bool:
 						params.Add(tags[0], strconv.FormatBool(v.Bool()))
+					case reflect.Ptr:
+						pv := v.Elem()
+						switch pv.Kind() {
```

Comment:
This is good, but to get all the `Kind`s without duplicating code, we should be able to move this to the top of the `switch` and add a label:
```go
loop:
  switch v.Kind() {
    case reflect.Ptr:
      v = v.Elem()
      goto loop
    case reflect.String:
    ...
}
```